lib/galaxy/model/__init__.py

```diff
@@ -5297,34 +5297,49 @@ def to_dict(self, view='collection', value_mapper=None, step_details=False, lega
             rval['steps'] = steps
 
             inputs = {}
-            for step in self.steps:
-                if step.workflow_step.type == 'tool':
-                    for job in step.jobs:
-                        if job is None:
-                            continue
-                        for step_input in step.workflow_step.input_connections:
-                            output_step_type = step_input.output_step.type
-                            if output_step_type in ['data_input', 'data_collection_input']:
-                                src = "hda" if output_step_type == 'data_input' else 'hdca'
-                                for job_input in job.input_datasets:
-                                    if not job_input.dataset:
-                                        continue
-                                    if job_input.name == step_input.input_name:
-                                        inputs[str(step_input.output_step.order_index)] = {
-                                            "id": job_input.dataset_id, "src": src,
-                                            "uuid" : str(job_input.dataset.dataset.uuid) if job_input.dataset.dataset.uuid is not None else None
-                                        }
+            for input_item_association in self.input_datasets + self.input_dataset_collections:
```

Comment:
❤️ 

Last time I changed something like this I did break some people's code I think (see ``legacy_job_state`` in the code base), but what you're doing here is a much better thing to do IMO. I love this.